posts/2022-02-24-Rust-1.59.0.md

```diff
@@ -0,0 +1,242 @@
+to the target architecture. Today, the following architectures are supported:
+
+* x86 and x86-64
+* ARM
+* AArch64
+* RISC-V
+
+You can see more examples of inline assembly in [Rust By Example][asm-example],
+and find more detailed documentation in the [reference][asm-reference].
+
+[asm-example]: https://doc.rust-lang.org/nightly/rust-by-example/unsafe/asm.html
+[asm-reference]: https://doc.rust-lang.org/nightly/reference/inline-assembly.html
+
+### Destructuring assignments
+
+You can now use tuple, slice, and struct patterns as the left hand side of an
+assignment.
+
+```rust
+let (a, b, c, d, e);
+
+(a, b) = (1, 2);
+[c, .., d, _] = [1, 2, 3, 4, 5];
+Struct { e, .. } = Struct { e: 5, f: 3 };
+
+assert_eq!([1, 2, 1, 4, 5], [a, b, c, d, e]);
+```
+
+This makes assignment more consistent with let bindings, which allow you to do
+the same thing. Note that assignments with operators such as `+=` are not
+allowed.
+
+### Const generics defaults and interleaving
+
+Generic types can now specify default values for their const generics. For
+example, you can now write the following:
+
+```rust
+struct ArrayStorage<T, const N: usize = 2> {
+    arr: [T; N],
+}
+
+impl<T> ArrayStorage<T> {
+    fn new(a: T, b: T) -> ArrayStorage<T> {
+        ArrayStorage {
+            arr: [a, b],
+        }
+    }
+}
+```
+
+Previously, type parameters were required to come before all const parameters.
+That restriction has been relaxed and you can now interleave them.
+
+```rust
+fn cartesian_product<
+    T, const N: usize,
+    U, const M: usize,
+    V, F
+>(a: [T; N], b: [U; M]) -> [[V; N]; M]
+where
+    F: FnMut(&T, &U) -> V
+{
+    // ...
+}
+```
+
+### Future incompatibility warnings
+
+Sometimes bugs in the Rust compiler cause it to accept code that should not
+have been accepted. An example of this was [borrows of packed struct
+fields][packed_borrows] being allowed in safe code.
+
+[packed_borrows]: https://github.com/rust-lang/rust/issues/46043
+
+While this happens very rarely, it can be quite disruptive when a crate used by
+your project has code that will no longer be allowed. In fact, you might not
+notice until your project inexplicably stops building!
+
+cargo now shows you warnings when a dependency will be rejected by a future
+version of Rust. After running `cargo build` or `cargo check`, you might see:
+
+```
+warning: the following packages contain code that will be rejected by a future version of Rust: old_dep v0.1.0
+note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
+```
+
+You can run the `cargo report` command mentioned in the warning to see a full
+report of the code that will be rejected. This gives you time to upgrade your
+dependency before it breaks your build.
+
+### Creating stripped binaries
+
+It's often useful to strip unnecessary information like debuginfo from binaries
+you distribute, making them smaller.
```

Comment:
Maybe add something about std shipping with debuginfo = 1 by default or something, which is why release builds may benefit from this even if they don't themselves turn on debug?

Reply:
Done in a suggestion below.